packages/sdk/src/types/runtime/dialog.ts

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { EventEmitter } from 'events';
+
+import { Context, GroupMask, Texture, User } from '.';
+
+/** The types of dialog boxes */
+export enum DialogType {
+	/** A dialog with text, an icon, and an OK button */
+	Message = 0,
+	/** A dialog with text, an icon, and OK and Cancel buttons */
+	Confirm,
+	/** A dialog with prompt text, a text input field, and OK and Cancel buttons */
+	Prompt
+}
+
+
+export enum DialogIcon {
+	/** A speech bubble */
+	Info = 0,
+	/** A yellow exclamation point */
+	Warning,
+	/** A red exclamation point */
+	Error,
+	/** A question mark */
+	Question
+}
+
+/** Pop a message to users, and get a response */
+export class Dialog extends EventEmitter {
```

Comment:
I think a fluent API design makes sense here:
```ts
const dialog = new Dialog(this.context)
    .type('info')
    .text("You lost the game.")
    .recipients(loser)
    .on('response', () => {
       console.log('User acknowledged.');
    });

dialog.show();
```

And a convenience method to avoid having to call the constructor:

```ts
Dialog.Create(this.context)
    .type('info')
    .text("You lost the game.")
    .recipients(loser)
    .on('response', () => {
       console.log('User acknowledged.');
    })
    .show();
```

Reply:
I'd rather have an initializer object, there's more precedent for it. In terms on in-lining handlers, you can already do that:
```ts
const dialog = new Dialog(this.context, {
    type: 'message',
    icon: 'info',
    text: "You lost the game",
    recipients: loser
})
.on('response', () => console.log('User acknowelged'))
.show();
```

---
In reply to: [320498358](https://github.com/microsoft/mixed-reality-extension-sdk/pull/444#discussion_r320498358) [](ancestors = 320498358)

Reply:
You have an unused var (`dialog`) in your example. This would trigger a lint error. You could work around it with parens, but that's just fighting the linter. A static `Create` method would solve it more cleanly. I had a comment to this effect in an earlier iteration, but don't see it now. Weird.